File: backend/tests/scripts/pg_test.js
/**
 * PostgreSQL Test Script
 * 
 * This script demonstrates how to execute queries using the sandbox `db` object.
 * The sandbox provides a pre-connected database wrapper - no need for require('pg').
 * 
 * Available globals:
 * - db.query(sql, params) - Execute PostgreSQL queries
 * - console.log/error/warn - Output capture
 * - JSON, Math, Date, Array, Object, String, Number - Built-in globals
 */

async function main() {
    console.log('=== PostgreSQL Test Script ===');
    console.log('Starting test queries...');

    // 1. Simple SELECT
    const timeResult = await db.query('SELECT NOW() as current_time, current_database() as db_name');
    console.log('Current time and database:', JSON.stringify(timeResult.rows, null, 2));

    // 2. Check PostgreSQL version
    const versionResult = await db.query('SELECT version()');
    console.log('PostgreSQL version:', versionResult.rows[0].version);

    // 3. List all tables in public schema
    const tablesResult = await db.query(`
    SELECT table_name 
    FROM information_schema.tables 
    WHERE table_schema = 'public' 
    ORDER BY table_name
  `);
    console.log('Tables in database:', tablesResult.rows.map(r => r.table_name));

    console.log('=== Test Complete ===');
}

main();

File: backend/tests/scripts/pg_test.py
"""
PostgreSQL Test Script (Python)

Demonstrates how to execute PostgreSQL queries using the sandbox `db` object.
The sandbox provides a pre-connected database wrapper.

Available globals:
    - db.query(sql, params) - Execute query and return results
    - db.execute(sql, params) - Execute without returning (INSERT/UPDATE/DELETE)
    - print() - Output capture
    - json, datetime, re, math - Allowed modules

USAGE:
    Submit via API with .py extension or scriptLanguage: 'python'
"""

# Simple SELECT query
print("=== PostgreSQL Python Test ===")
print("Starting test queries...")

# 1. Get current time and database
result = db.query('SELECT NOW() as current_time, current_database() as db_name')
print(f"Current time: {result['rows'][0]['current_time']}")
print(f"Database: {result['rows'][0]['db_name']}")

# 2. Get PostgreSQL version
version_result = db.query('SELECT version()')
print(f"PostgreSQL version: {version_result['rows'][0]['version'][:50]}...")

# 3. List tables in public schema
tables_result = db.query("""
    SELECT table_name 
    FROM information_schema.tables 
    WHERE table_schema = 'public' 
    ORDER BY table_name
    LIMIT 10
""")
print(f"Tables found: {len(tables_result['rows'])}")
for row in tables_result['rows'][:5]:
    print(f"  - {row['table_name']}")

# 4. Parameterized query example
count_result = db.query(
    "SELECT COUNT(*) as count FROM information_schema.tables WHERE table_schema = %s",
    ('public',)
)
print(f"Total public tables: {count_result['rows'][0]['count']}")

print("=== Test Complete ===")
